### PR TITLE
Change to trigger CI

### PR DESCRIPTION
**Pull request by maxi297**
## What
Follow-up on https://github.com/airbytehq/airbyte/pull/40713, addresses https://github.com/airbytehq/airbyte-internal-issues/issues/8447

On CDK changes that can modify the binaries, run source-shopify tests.

## How

By adding a new GitHub workflow that will get triggered if there are any changes on the following files
 - 'airbyte-cdk/python/airbyte_cdk/**'
 - 'airbyte-cdk/python/bin/**'
- 'airbyte-cdk/python/poetry.lock'
- 'airbyte-cdk/python/pyproject.toml'

Manual tests performed:
* The CI check is marked as green in 4 seconds when the pull request does not include CDK changes
* Change in the CDK trigger testing for source-shopify
* (currently not behaving as expected (see print screen)) Pushing a new change stops the current tests
image

Things I haven't tested and assume works:
* use-local-cdk flag

## User Impact
The devs will now know if their CDK changes are working for source shopify. Eventually, we will add more connectors to that

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by natikgadzhi — APPROVED** [review]
Approving to unblock. Are you _sure_ you want this on every change in every PR instead of manual, though? Sounds like for quite a lot of small changes you don't really want to run the whole battery?

File: .github/workflows/cdk_connectors_tests.yml [hunk 1]
```diff
@@ -54,11 +53,11 @@ jobs:
             }' \
 
   connectors_ci:
-    needs: changes
+    needs: cdk_changes
     # We only run the Connectors CI job if there are changes to the connectors on a non-forked PR
     # Forked PRs are handled by the community_ci.yml workflow
     # If the condition is not met the job will be skipped (it will not fail)
-    if: github.event_name == 'pull_request' && needs.changes.outputs.python_cdk == 'true' && github.event.pull_request.head.repo.fork != true
+    if: github.event_name == 'pull_request' && needs.cdk_changes.outputs.python_cdk == 'true' && github.event.pull_request.head.repo.fork != true
```
Comment (alafanechere) [inline .github/workflows/cdk_connectors_tests.yml:60]: You still have `workflow_dispatch` trigger. It's usually handy to keep it in order to manually run the workflow. Your `if` here will not run the `connectors_ci` job on workflow dispatch...

**Review by alafanechere — APPROVED** [review]
LGTM!

About _Pushing a new change stops the current tests_ : 
You concurrency group definition look correct, so I don't really know what's happenning.
Graceful cancellation of the previous workflow might take time... 🤔 I'd suggest you try out to set the [concurrency at the job level](https://docs.github.com/en/enterprise-cloud@latest/actions/using-jobs/using-concurrency#using-concurrency-in-different-scenarios), you might get better results.

**Review by clnoll — APPROVED** [review]
LGTM but I second @natikgadzhi's question - 

> Are you sure you want this on every change in every PR instead of manual, though? Sounds like for quite a lot of small changes you don't really want to run the whole battery?

Especially if you do this for more sources.

**Comment by maxi297** [comment]
> Sounds like for quite a lot of small changes you don't really want to run the whole battery?

@natikgadzhi @clnoll Can you share a bit more here as to what is the concern? What would be the downside of running this on every CDK change? Is it a resource issue?

**Comment by clnoll** [comment]
Assuming that CDK changes happen more frequently than your average connector change, I'd be concerned with both resource usage and API rate limits against our sandbox accounts. Could be annoying if we end up with a massive queue of jobs and/or transient failures.

That said I'd be fine waiting to see if these (or other) issues actually happen before we worry about it.

**Comment by maxi297** [comment]
@clnoll That is a good point. We are basically already at risk for that but we will be even more with that. 

> Assuming that CDK changes happen more frequently than your average connector change

We indeed have more changes in the CDK. Do those changes happen at the same time we modify connectors? We will definitely increase the risk of that but I'm not worried about this yet. If we hit that, I would argue that we could reduce the dataset for those connectors and if needed move some of the coverage of those tests to mock server tests. Let's see how it goes.

For resources, how do we track that and at which point isn't it worth it anymore?

**Comment by clnoll** [comment]
> We indeed have more changes in the CDK. Do those changes happen at the same time we modify connectors? We will definitely increase the risk of that but I'm not worried about this yet. If we hit that, I would argue that we could reduce the dataset for those connectors and if needed move some of the coverage of those tests to mock server tests. Let's see how it goes.

Waiting to see how it goes makes sense to me.

> For resources, how do we track that and at which point isn't it worth it anymore?

I think what we'd want is to know how long jobs are spending in the `queued` state, and whether we see an increase in that once your PR is deployed. However, I'm not sure if we're currently tracking that. @alafanechere is that something you know?

**Comment by alafanechere** [comment]
> > We indeed have more changes in the CDK. Do those changes happen at the same time we modify connectors? We will definitely increase the risk of that but I'm not worried about this yet. If we hit that, I would argue that we could reduce the dataset for those connectors and if needed move some of the coverage of those tests to mock server tests. Let's see how it goes.
> 
> Waiting to see how it goes makes sense to me.
> 
> > For resources, how do we track that and at which point isn't it worth it anymore?
> 
> I think what we'd want is to know how long jobs are spending in the `queued` state, and whether we see an increase in that once your PR is deployed. However, I'm not sure if we're currently tracking that. @alafanechere is that something you know?

We are not tracking that.

I did not raise concerns about the CI load or rate limiting issue induced by this change as @maxi297 will be the first one to suffer from it if it happens and will be incentivized to find workarounds.

**Comment by natikgadzhi** [comment]
Don’t get me wrong, it’s ok to cross this bridge when we get there.

Maybe there is a subsection of cdk files that we care about? Ie only test
automatically on some changes that are particularly prone to problems.

But then again, subtle change in jinja security context is an example of a
change that I would not suspect.

File: .github/workflows/cdk_connectors_tests.yml (hunk 1 above)
Comment (maxi297) [reply .github/workflows/cdk_connectors_tests.yml]: Checking this error, I was checking the other workflows and it feels like [this](https://github.com/airbytehq/airbyte/blob/4e9146425b00d98b9d144d71def7e2496251d3de/.github/workflows/connectors_tests.yml#L87-L90) is not used. I provided a fix for that too

**Comment by maxi297** [comment]
> You concurrency group definition look correct, so I don't really know what's happenning. Graceful cancellation of the previous workflow might take time... 🤔 I'd suggest you try out to set the [concurrency at the job level](https://docs.github.com/en/enterprise-cloud@latest/actions/using-jobs/using-concurrency#using-concurrency-in-different-scenarios), you might get better results.

I experienced some slowness on https://github.com/airbytehq/airbyte-platform-internal/pull/13023 too so I would assume this might be a GitHub issue and not a workflow issue. I'll assume this will work for now and ask people to be vocal if they see any issue

**Pull request merged**